### ISSUE TITLE
Validate all objects referenced in scripts at parsing stage

### ISSUE THREAD

**Issue by balloonpopper** [godot-escoria/escoria-demo-game#1008]
**Is your feature request related to a problem? Please describe.**
I just had an error come up working on room 10 where the "walk_block" command was used. When the script executed it was trying to run this command : 
```
walk_block player accept_input_location
```
except that the location "accept_input_location" didn't exist in my version of the room, so when Escoria got to that line in the script it errored with 
```
Errors in file walk_block: invalid second object
Object with global id accept_input_location not found
```

I'd like to put forward a suggestion that once all the room scripts have been parsed, there's a second parsing round that checks the validity of all the objects. If Escoria knows the content of the scripts, and knows all the ESCitems in the room, is there a reason it can't check that the scripts are only referencing objects that exist, and errors when it finds an invalid one?

Checking this automatically has the potential to save a lot of time tracking down bugs in scripts.

**Describe the solution you'd like**
As above

**Describe alternatives you've considered**
None

**Additional context**
I appreciate this might not be feasible for multiple reasons 
- it might slow down room loading too much
- it would need a solution to account for things like creating ESCitems/locations etc in code as a result of something that happens while playing a room (e.g. opening a magic chest gives you a random object which is only instantiated when needed)
- it might be too hard based on how everything's structured

**Comment by BHSDuncan** [comment]
@StraToN this might be a non-issue now that we've moved to ASHES. Thoughts?

**Comment by StraToN** [comment]
Indeed. Parsing ASHES scripts is now possible so this issue can be now closed. Certainly more things can be done to improve even further, but this one is ok to be. Closing then.
